components/backdrop/backdrop.js

```diff
@@ -162,20 +162,12 @@ function hideAccessible(target) {
 			if (path.indexOf(child) !== -1) continue;
 			if (child.hasAttribute(BACKDROP_HIDDEN)) continue;
 
-			const role = child.getAttribute('role');
-			if (role) child.setAttribute(BACKDROP_ROLE, role);
-			child.setAttribute('role', 'presentation');
-
-			if (child.nodeName === 'FORM' || child.nodeName === 'A') {
```

Comment:
I vaguely recall discussing this code block before and questions around why exactly this was only targeting forms and links... it seems like it [was introduced in 2013 into our MVC dialog code](https://search.d2l.dev/diff/lms/lp/framework/web/D2L.LP.Web.UI/Desktop/Controls/Shim/Shim.js?r1=%2Flms%2Flp%2Fframework%2Fweb%2FD2L.LP.Web.UI%2FDesktop%2FControls%2FShim%2FShim.js%404130e589&r2=%2Flms%2Flp%2Fframework%2Fweb%2FD2L.LP.Web.UI%2FDesktop%2FControls%2FShim%2FShim.js%40186e88e1) with no comment about why unfortunately.

I _think_ we should be OK to do this, but it is mildly worrying especially if this is going to be hotfixed.

Reply:
I think [this PR](https://github.com/BrightspaceUI/core/pull/2988) is what you're referring to. 

I played around with VO on the old backdrop and I was still able to focus elements that had the `role` but no `aria-hidden` so maybe it was added to prevent those from being focused. But then why wouldn't you just apply it to everything?

Reply:
Thanks for finding that -- haha I guess this is us circling back!

Ok I sleuthed through the old Bitbucket archive and found where this [was first introduced](https://old-git.d2l.dev/projects/CORE/repos/lp/commits/bf730c5fa89badc28d012ffa1df721bd6256ee8e#framework/web/D2L.LP.Web.UI/Desktop/MasterPages/Dialog/Default/Dialog.js). The commit message is:
> When a dialog shims the main page, ensure that form controls have
aria-hidden=true set, to prevent AT such as JAWS or NVDA from passing control to
shimmed form elements. Closing the dialog properly resets the aria-hidden
attribute.

So that makes sense, and eases my concerns a bit. The code was [then moved into Shim.js here](Reduce number of H1 elements when modal dialogs are open.
This aligns the code in MVC and Legacy dialogs to ensure that main page elements
are set to aria presentation mode when a dialog is shimmed.
Legacy dialogs will now also use an h2 element instead of h1 for the d2l-heading-title element.) and that's where the `<a>` code was added. I'll dig further and see if I can find the old legacy version of this and see if there are more clues.

Reply:
That's lucky that they added such a detailed commit message 😄 

Thanks for finding this

Reply:
Unfortunately I hit a brick wall on the `<a>` stuff. The closest I came was [this change](https://old-git.d2l.dev/projects/CORE/repos/lp/commits/f13f2fe0632cbf1efc558037b3db02383597b3a7#framework/web/D2L.Web/ui/Dialog/03_ModalDialog.js), which switched legacy dialogs to use the MVC shim code in 2013. But that [legacy shim code](https://old-git.d2l.dev/projects/CORE/repos/lp/browse/framework/web/D2L.Web/ui/page/JavaScript/ShimController.js) doesn't do anything with ARIA... so yeah, dead end.

Reply:
Yeah, I suspected it was something related to aria-hidden support. Thanks for searching into this, I'll continue to test it to make sure everything is working as expected.